src/include/migraphx/generate.hpp

```diff
@@ -36,36 +36,38 @@ inline namespace MIGRAPHX_INLINE_NS {
 template <class T, MIGRAPHX_REQUIRES(is_floating_point<T>{})>
 constexpr T normalize(unsigned long z)
 {
-    if(z == 0)
-        return T(0);
-    const auto max     = 32;
-    const double range = max / 2; // NOLINT
-    double result      = double(z % max) / range;
-    result -= 1;
+    const auto max     = 1ULL << (sizeof(T) * 8 - 1);
+    const double range = max / 2.0;
+    double result      = -1.0 + (z % max) / range;
     return T(result);
+    // Expected output: between -1.0 and 1.0
 }
 
 template <class T, MIGRAPHX_REQUIRES(is_signed<T>{} and not is_floating_point<T>{})>
 constexpr T normalize(unsigned long z)
 {
-    const auto max      = 1ULL << (sizeof(T) * 5);
+    const long long max = 1ULL << (sizeof(T) * 8 - 2);
     const auto half_max = max / 2;
-    return half_max - (z % max);
+    auto result         = half_max - (z % max);
+    return T(result);
+    // Expected output: between -half_max and half_max
 }
 
 template <class T,
           MIGRAPHX_REQUIRES(not is_signed<T>{} and std::is_integral<T>{} and
                             not std::is_same<T, bool>{})>
 constexpr T normalize(unsigned long z)
 {
-    const auto max = 1ULL << (sizeof(T) * 5);
+    const auto max = 1ULL << (sizeof(T) * 8 - 1);
```

Comment:
Same here, this should not be changed.

Reply:
Add an inline comment explaining why the value exists.

Reply:
Comment added to explain why a certain range is chosen. Thanks.